doc/source/report/classification.rst

```diff
@@ -76,83 +88,240 @@ selection. The process will also help us to find outliers and missing
+        mouse_day_X=mouse_dayavgstd_rsl[:,2:],
+        method=method, dist=dist, num_clusters=range(2,17))
+
+    # plot 
+    plot_clustering.plot_dendrogram(
+        mouse_day=mouse_dayavgstd_rsl, method=method, dist=dist)
+
+
+Testing Framework Outline
 -------------------------
 
--  The first step to test the reproducibility is to test the stability
-   of classification models. Since we randomly split the dataset to be
-   the test set and the training set, we can train and test the model
-   over different seeds and plot the accuracy against different
-   trials. We should also see if the important features are stable over
-   different trials.
-
--  From our limited understanding, the results of this research might
-   have a meaningful implication on the way we treat psychological
-   disorders. If it turns out that nature does influence these
-   disorders, we can probably conclude that psychological disorders is
-   not much different than physical disabilities. Otherwise, if nature
-   has little influence over these disorders, we can try to find way to
-   prevent these disorders from happening.
-
-Initial tasks
+To ensure our functions do the correct steps and return appropriate 
+results, we also implemented test functions.
+For clustering, we first perform basic testing of whether our output 
+has appropriate number of values or values we expect. One more advanced 
+check we perform is to test whether we successfully assign cluster numbers 
+to every observation. Also, since we compute silhouette score for each 
+cluster and silhouette score is defined to be between -1 and 1, we also 
+checked that whether our silhouette score is appropriate.
+For classification, we also checked whether our final predictions of mouse 
+strains only include numbers 0 through 15 since they are the only strains 
+for data we have and we should predict those strains.
+
+
+Result
 -------------
 
-1. Clean up the existing strain\_classification.py: create functions and
-   objects.
-2. Adding new models: knn, random forests, neural networks, logistic
-   regressions.
-3. Doing unsupervised learning: k-means.
-4. Compute confidence interval for the accuracy of each model.
-5. Give insights about how mice behavior is related to their genetic
-   heritage (strain difference).
+**Classification**
+
+For three models, after tuning the parameters and output the prediction result, we create the side-by-side barplot for the different measurement of accuracy, which are precision, recall and F1. 
+
+precision(P) = $\frac{\# label\ y\ \ predicted\ correctly}{\# label\ y\ predicted}$
+
+recall(R) = $\frac{\# label\ y\ \ predicted\ correctly}{\# label\ y\ true}$
+
+F-1 = $\frac{2*P*R}{P+R}$
+
+*Random Forest*
+
+Random Forest shows a very promising result. For each strain, prediction, recall and F-1 measure are very closed to each other. Except for predicting strain 15, all the other prediction has F-1 measure exceeding 0.8.
```

Comment:
"closed" should be "close"

"Except for strain 15, each prediction gave an F-1 measure exceeding 0.8.